core/src/commonMain/kotlin/com/revenuecat/purchases/kmp/PurchasesAreCompletedBy.kt

```diff
@@ -3,21 +3,41 @@ package com.revenuecat.purchases.kmp
 /**
  * Modes for completing the purchase process.
  */
-public expect enum class PurchasesAreCompletedBy {
+public sealed interface PurchasesAreCompletedBy {
+
     /**
-     * RevenueCat will automatically acknowledge verified purchases. No action is required by you.
+     * RevenueCat will automatically acknowledge verified purchases.
+     * No action is required by you.
      */
-    REVENUECAT,
+    public data object RevenueCat : PurchasesAreCompletedBy
```

Comment:
Hmm does this need to be a `data object`? I think it could just be an object, but lmk if I missed something!

Reply:
`data object` is the recommended type in `sealed` hierarchies, with a minor lint "suggestion" appearing if you omit it: 
![Scherm­afbeelding 2024-08-19 om 15 25 05](https://github.com/user-attachments/assets/24030fee-5c2c-4cce-95be-3b7e403b63e1)

Reasoning in [the docs](https://kotlinlang.org/docs/object-declarations.html#using-data-objects-with-sealed-hierarchies):
> [...] they allow you to maintain symmetry with any data classes you may have defined alongside the object. In this example, declaring `EndOfFile` as a `data object` instead of a plain `object` means that it will get the `toString()` function without the need to override it manually:

Reply:
Makes sense, thanks!